src/reduxTypes.js

```diff
@@ -153,9 +153,10 @@ export type FlagName = $Keys<FlagsState>;
  * See also `NarrowsState`, which is an index on this data that identifies
  * messages belonging to a given narrow.
  */
-export type MessagesState = {|
-  [id: number]: $Exact<Message>,
-|};
+// This type cannot be made exact until Flow v0.111.0.
+export type MessagesState = {
+  [id: number]: Message,
```

Comment:
Instead of removing exactness here, why not add it on `Message` itself? (And remove the then-redundant mention of it here.)

Otherwise we're regressing on how tightly we type-check.

For a quick experiment I made `Message` exact. Flow reported only one error -- and in fact it looks like a real one, which would have been caught in 3dc2d8fe7 if we'd had `Message` exact then.

Reply:
`Message` really *isn't* exact, though: it's full of unlisted fields from (possibly future versions of) the server. We shouldn't make the claim that it is. (The bug you cite is a good argument that `eg.streamMessage` and friends should take `$Exact<Message>`, though.)

I assume what we really want to do is to deconstruct the inexact, server-supplied `ApiMessage` object (to coin a name) into a fresh `LocalMessage` object that really is exact. But not, I think, today. 

(I'm pretty sure this has come up before. I know I've documented it in `Message` at some point, but I suppose that hasn't made it into a PR yet. I'll try to dig it up and submit it separately. Or just rewrite it.)